### PR TITLE
Add Wagtail 5.0 support

### LINKED ISSUE #73: Wagtail 5.0 not supported
When attempting to upgrade to Wagtail 5.0:

```wagtail-factories 4.0.0 depends on wagtail<5.0 and >=2.15```

### PR DESCRIPTION
**Pull request by jams2**
Closes #73 

Supercedes #69, #70

Based on #70, which adds support for Wagtail 4.2, this adds support for Wagtail 5.0.

### CODE REVIEW

File: tests/test_factories.py [hunk 1]
```diff
@@ -93,9 +88,8 @@ def test_custom_page_streamfield():
     assert len(page.body) == 0
 
 
-@pytest.mark.skipif(
-    wagtail_version >= (4, 1),
-    reason="Cannot assign a naive list to a ListBlock value if using ReferenceIndex",
+@pytest.mark.skip(
+    reason="Cannot assign a naive list to a ListBlock value if using ReferenceIndex"
```
Comment (zerolab) [inline tests/test_factories.py:92]: question: is this needed at all? previously it was skipped for Wagtail >= 4.1 and now we only support Wagtail 4.1+ 🤔

File: tox.ini [hunk 2]
```diff
@@ -1,10 +1,8 @@
 [tox]
 minversion = 3.14.6
 envlist =
-    py{37,38,39,310}-django32-wagtail{215,216}-factoryboy32
-    py{38,39,310}-django40-wagtail{216,30}-factoryboy32
-    py{38,39,310}-dj40-wagtail40-factoryboy32
-    py{39,310}-dj41-wagtail40-factoryboy32
+    py{37}-django{32}-wagtail{41,42,50}-factoryboy32
+    py{38,39,310,311}-django{32,40,41}-wagtail{41,42}-factoryboy32
```
Comment (zerolab) [inline tox.ini:5]: officially Django 3.2 supports up to Python 3.10
https://docs.djangoproject.com/en/3.2/releases/3.2/ but I suppose this is ok

File: tests/testapp/migrations/0003_alter_mytestpage_body_and_more.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
+# Generated by Django 4.0.10 on 2023-05-16 04:25
```
Comment (zerolab) [inline tests/testapp/migrations/0003_alter_mytestpage_body_and_more.py:1]: suggestion: regenerate the testapp migrations to have just the initial one

**Review by zerolab — COMMENTED** [review]
Left a couple of questions/suggestions. Nothing blocking

File: tests/test_factories.py (hunk 1 above)
Comment (jams2) [reply tests/test_factories.py]: Well spotted. I've changed it to test assignment of a `ListValue` to the `StreamField` which still has some value I think

File: tests/testapp/migrations/0003_alter_mytestpage_body_and_more.py (hunk 3 above)
Comment (jams2) [reply tests/testapp/migrations/0003_alter_mytestpage_body_and_more.py]: Done in https://github.com/wagtail/wagtail-factories/pull/74/commits/83a6015322ecd67e032b210f85abf3743d7d26e2

**Pull request merged**